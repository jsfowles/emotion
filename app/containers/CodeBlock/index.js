import React from 'react';
import PropTypes from 'prop-types';
import hljs from 'highlight.js';

import TypeScale from '@identity/type';

export default class CodeBlock extends React.PureComponent {
  static defaultProps = {
    language: '',
  };

  static propTypes = {
    value: PropTypes.string.isRequired,
    language: PropTypes.string,
  };

  componentDidMount() {
    this.highlightCode();
  }

  componentDidUpdate() {
    this.highlightCode();
  }

  setRef = (el) => {
    this.codeEl = el;
  };

  highlightCode() {
    hljs.highlightBlock(this.codeEl);
  }

  render() {
    return (
      <TypeScale.Pre>
        <code ref={ this.setRef } className={ this.props.language }>
          {this.props.value}
        </code>
      </TypeScale.Pre>
    );
  }
}
