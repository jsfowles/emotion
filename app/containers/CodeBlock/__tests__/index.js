import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CodeBlock from '@containers/CodeBlock';

const defaultProps = {
  value: `
  console.log("hello world");
  `,
  language: 'js',
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };

  return <CodeBlock { ...newProps } />;
};

describe('CodeBlock', () => {
  it('renders props', () => {
    const wrapper = mount(generateComponent());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
