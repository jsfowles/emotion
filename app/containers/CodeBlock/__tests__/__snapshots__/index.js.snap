// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock renders props 1`] = `
<CodeBlock
  language="js"
  value="
  console.log(\\"hello world\\");
  "
>
  <glamorous(pre)>
    <pre
      className="css-drwtoh"
    >
      <code
        className="js"
      >
        
  console.log("hello world");
  
      </code>
    </pre>
  </glamorous(pre)>
</CodeBlock>
`;
